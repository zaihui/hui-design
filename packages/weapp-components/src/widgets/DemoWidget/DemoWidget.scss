.demo-widget {
  border: solid 1px #000;
}
