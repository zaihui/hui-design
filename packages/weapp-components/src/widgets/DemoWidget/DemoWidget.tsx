import React from 'react'
import { View } from '@tarojs/components'

export interface DemoWidgetProps {
  /**
   * 一个字符串型的属性
   */
  prop1: string
  /**
   * 一个数值类型的可选属性
   */
  prop2?: number
}
/**
 * 一个示例Widget
 */
export default class LoginWidget extends React.Component<DemoWidgetProps> {
  // eslint-disable-next-line class-methods-use-this
  render(): JSX.Element {
    const { prop1, prop2 = 0 } = this.props
    return (
      <View className='demo-widget'>
        this is DemoWidget
        <View>prop1 is {prop1}</View>
        <view>prop2 is {prop2}</view>
      </View>
    )
  }
}
