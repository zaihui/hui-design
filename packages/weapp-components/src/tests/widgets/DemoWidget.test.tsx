import React from 'react'

import { renderToString } from 'nerv-server'
import DemoWidget from '@/widgets/DemoWidget/DemoWidget'

describe('demoWidget Snap', () => {
  it('render DemoWidget without prop2', () => {
    expect.hasAssertions()
    expect(renderToString(<DemoWidget prop1='foo' />)).toMatchSnapshot()
  })
  it('render DemoWidget with prop2', () => {
    expect.hasAssertions()
    expect(renderToString(<DemoWidget prop1='foo' prop2={233} />)).toMatchSnapshot()
  })
})
