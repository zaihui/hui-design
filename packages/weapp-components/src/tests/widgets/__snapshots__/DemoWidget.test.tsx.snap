// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demoWidget Snap render DemoWidget with prop2 1`] = `"<div class=\\"demo-widget\\">this is DemoWidget<div class=\\"\\">prop1 is foo</div><view>prop2 is 233</view></div>"`;

exports[`demoWidget Snap render DemoWidget without prop2 1`] = `"<div class=\\"demo-widget\\">this is DemoWidget<div class=\\"\\">prop1 is foo</div><view>prop2 is 0</view></div>"`;
